### PR TITLE
feat: use new engine in e2e tests for OP

### PR DESCRIPTION
**Pull request by klkvr**
Defaults to new engine when running op e2e tests.

This required some changes to a test

### CODE REVIEW

File: crates/optimism/node/tests/e2e/p2p.rs [hunk 1]
```diff
@@ -77,25 +83,9 @@ async fn can_sync() -> eyre::Result<()> {
         .wait_block(
             side_payload_chain[0].0.block().number,
             side_payload_chain[0].0.block().hash(),
-            true,
+            false,
```
Comment (klkvr) [inline crates/optimism/node/tests/e2e/p2p.rs:86]: this marks whether we need to wait for `Finish` stage to advance which might not happen if persistence threshold is not reached

File: crates/optimism/node/tests/e2e/p2p.rs [hunk 2]
```diff
@@ -43,13 +43,19 @@ async fn can_sync() -> eyre::Result<()> {
         .update_optimistic_forkchoice(canonical_chain[tip_index - reorg_depth])
         .await?;
     second_node
-        .wait_block((tip - reorg_depth) as u64, canonical_chain[tip_index - reorg_depth], true)
+        .wait_block((tip - reorg_depth) as u64, canonical_chain[tip_index - reorg_depth], false)
```
Comment (klkvr) [inline crates/optimism/node/tests/e2e/p2p.rs:46]: this marks whether we need to wait for `Finish` stage to advance which might not happen if persistence threshold is not reached

File: crates/optimism/node/tests/e2e/p2p.rs [hunk 3]
```diff
@@ -43,13 +43,19 @@ async fn can_sync() -> eyre::Result<()> {
...
         .await?;
     second_node.engine_api.canonical_stream.next().await.unwrap();
 
-    // On third node, sync optimistically up to block number 90a
+    // Trigger backfil sync until block 80
```
Comment (klkvr) [inline crates/optimism/node/tests/e2e/p2p.rs:50]: This seems to be an edge case for new engine. If we sync to the tip via backfill sync and then a reorg happens, we're not going to be able to process it because we will be missing trie updates in in-memory state.

I guess this is unlikely because we'd mostly use backfill sync to reach finalized block

**Review by mattsse — APPROVED** [review]
